src/Microsoft.OData.Core/UriParser/Parsers/ODataPathParser.cs

```diff
@@ -1183,6 +1269,35 @@ private void CreateNextSegment(string text)
             this.CreateDynamicPathSegment(previous, identifier, parenthesisExpression);
         }
 
+        private bool TryCreateEscapeFunctionSegment(string segmentText)
+        {
+            int numberOfSegmentsParsed = this.parsedSegments.Count;

```

Comment:
>int numberOfSegmentsParsed = this.parsedSegments.Count; [](start = 12, length = 55)

line 1274 can move to Line 1290, right?

Reply:
Nope. BindSegmentForEscapeFunction changes the list of segments and it's count. 

---
In reply to: [369798577](https://github.com/OData/odata.net/pull/1621#discussion_r369798577) [](ancestors = 369798577)